### DISCUSSION TITLE
Breaking changes in Bolt 5.0

### DISCUSSION THREAD

**Discussion by bobdenotter** [bolt/core#2556] [General]
This is a list of breaking changes in Bolt 5.0, that we'll need to document when 5.0 is released: 

## The `x-powered-by: Bolt` header

The setting for this has changed from:

```yaml
omit_powered_by_header: false
```

to:

```yaml
headers:
    powered_by: true
```

**Comment by I-Valchev** [comment]
@bobdenotter 
https://github.com/bolt/core/blob/4.1/src/Extension/BaseExtension.php#L132 <- this is also a breaking change.

One known usage so far: https://github.com/bolt/newswidget/blob/master/src/Extension.php

**Comment by bobdenotter** [comment]
People will need to update their `public/index.php`, `.gitignore` and `bin/console` files. If they don't, configuration in `.env.local` will not be used. 

https://symfony.com/doc/current/configuration/dot-env-changes.html

These are the correct files: 

 - https://github.com/bolt/project/blob/master/.gitignore
 - https://github.com/bolt/project/blob/master/public/index.php
 - https://github.com/bolt/project/blob/master/bin/console
